tools/clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -749,9 +812,62 @@ class ConstExprEmitter :
     case CK_HLSLCC_IntegralToFloating:
     case CK_HLSLCC_FloatingToIntegral:
     case CK_HLSLCC_FloatingToBoolean:
-      // Since these cast kinds have already been handled in ExprConstant.cpp,
-      // we can reuse the logic there.
-      return CGM.EmitConstantExpr(E, E->getType(), CGF);
+      if (hlsl::IsHLSLMatType(E->getType()) && hlsl::IsHLSLMatType(E->getSubExpr()->getType()) && C) {
```

Comment:
I don't think C can be null here because of the ` if (!C) return nullptr;` line above